src/test/java/com/hermesworld/ais/galapagos/notifications/impl/NotificationEventListenerTest.java

```diff
@@ -84,4 +132,21 @@ private TopicEvent buildTestEvent(String envId) {
         return new TopicEvent(context, metadata);
     }
 
+    private static class MockCluster {
+
+        private final KafkaCluster cluster;
+
+        private final Map<String, TopicBasedRepositoryMock<?>> repositories = new HashMap<>();
+
+        public MockCluster(String id) {
+            cluster = mock(KafkaCluster.class);
+            when(cluster.getId()).thenReturn(id);
+            when(cluster.getRepository(any(), any())).then(
+                    inv -> repositories.computeIfAbsent(inv.getArgument(0), key -> new TopicBasedRepositoryMock<>()));
+        }
+
+        public KafkaCluster getCluster() {
+            return cluster;
+        }
+    }
```

Comment:
Why do you need your own Mock class here? It only seems to be a holder for a mocked Cluster object, and has no value on its own.